catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -59,55 +60,72 @@
 
     private int subDirLevels;
 
+    private PeriodicBatchExecutor<Metacard> periodicDrain;
+
     private boolean enableBackupPlugin = true;
 
     public CatalogBackupPlugin() {
+
+        init();
+    }
+
+    private void init() {
         subDirLevels = 0;
+        setPeriodicDrain(new PeriodicBatchExecutor<>(1000, 60, TimeUnit.SECONDS));
+        getPeriodicDrain().execute(this::backup);
     }
 
     /**
      * Backs up created metacards to the file system backup.
      *
      * @param input the {@link CreateResponse} to process
-     * @return
+     * @return {@link CreateResponse}
      * @throws PluginExecutionException
      */
     @Override
-    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+    public synchronized CreateResponse process(CreateResponse input)
```

Comment:
I hope that isn't the case. I'd have to change the executor's interface. Right now, the executor accepts instances of some object (in this case Metacards) and then batches them up for processing. The system works because there is only one operation-- backup. To handle multiple operations, I would switch the executor to accept instances of Runnable or some functional interface, so I could pair up a task (backup, update, etc) with the instance of Metacard.No more batching things up. 

That wouldn't be so bad, really. Ultimately,  a file  is opened for each Metacard, so even if the batches get small, it might not change the runtime much-- the number of file access stays the  same.  I'll ask @stustison to weigh in.